cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/instance/DatasetInstanceManager.java

```diff
@@ -97,12 +98,22 @@ public DatasetSpecification call() throws Exception {
    * @return collection of {@link DatasetSpecification} of all dataset instances in the given namespace
    */
   public Collection<DatasetSpecification> getAll(final NamespaceId namespaceId) {
+    return getAll(namespaceId, new HashMap<String, String>());
+  }
+
+  /**
+   * @param namespaceId {@link NamespaceId} for which dataset instances are required
+   * @param properties {@link Map} of dataset properties
+   * @return collection of {@link DatasetSpecification} of all dataset instances in the given namespace which
+   * are having the specified properties
+   */
+  public Collection<DatasetSpecification> getAll(final NamespaceId namespaceId, final Map<String, String> properties) {
```

Comment:
Should call it `get` or `getSpecifications`.

Reply:
renamed to `get`